Assets/Scripts/Enemy Scripts/EnemyBehavior.cs

```diff
@@ -45,79 +49,178 @@ public class EnemyBehavior : MonoBehaviour
     [Tooltip("Layer on which to check for other enemies. (LayerMask)")]
     private LayerMask enemyLayer;
 
+    [SerializeField]
+    [Tooltip("The maximum distance the enemy will fly from its original position before turning around.")]
+    private float maximumFlightDistance = 10;
+
+    [Tooltip("The value of this enemy's initial x-position.")]
+    private float startingPositionX;
+
     [Tooltip("True = enemy can shoot, False = enemy can't shoot.")]
     private bool canShoot = true;
+
+    [Tooltip("True = enemy can switch which direction it's facing, False = enemy cannot do this.")]
+    private bool canFlip = true;
+
+    [SerializeField]
+    [Tooltip("The minimum amount of time an enemy must wait since the last time it flipped directions before flipping again.")]
+    private float flipCooldown = 0.1f;
+
     /// <summary>
     /// True = enemy is facing right, False = enemy is facing left
     /// </summary>
     internal bool FacingRight { get; private set; }
     /// <summary>
-    /// True = enemy is aiming upward, False = enemy is not aiming upward
+    /// Returns true if this enemy is a Flier type, otherwise returns false. (Readonly)
     /// </summary>
-    internal bool AimingUp { get; private set; }
+    internal bool EnemyTypeIsFlier
+    {
+        get { return IsFlier(); }
+    }
 
     [Tooltip("The Rigidbody2D component attached to the Enemy prefab.")]
     private Rigidbody2D rigidbody2D;
     [Tooltip("The SpriteRenderer component attached to the Enemy prefab.")]
     private SpriteRenderer spriteRenderer;
 
+    [Tooltip("The Jetpack game object that may or may not exist as a child of this enemy game object, if this enemy is a Flier.")]
+    private GameObject jetpack;
+
     private void Start()
+    {
+        AssignComponents();
+        SetFacingRightToTrue();
+        GetStartingPositionX();
+    }
+    /// <summary>
+    /// Let the script know that the enemy is facing right. (Note: This necessitates that all enemies begin the level facing right)
+    /// </summary>
+    private void SetFacingRightToTrue()
+    {
+        FacingRight = true;
+    }
+    /// <summary>
+    /// Get this enemy game object's x-position.
+    /// </summary>
+    private void GetStartingPositionX()
+    {
+        startingPositionX = gameObject.transform.position.x;
```

Comment:
You can kind of see how properties correlate with this.

A smart way to do this would be to just remove this field and make it a private read only property like this:

`private float StartingPositionX => gameObject.transform.position.x;`